TypeScript compilation fails due to missing type deps

## Software versions

* **OS**: Ubuntu 16.04.3 LTS
* **Consumer Pact library**: PactJS 5.3.1
* **Node Version**: v6.9.1

## Expected behaviour
When using pact-js as a `devDependency`, pact-js can be compiled by TypeScript successfully without requiring additional definitions (e.g. adding `es6-promises` as an explicit dependency).

## Actual behaviour
TypeScript compilation fails.

## Steps to reproduce

 1. Clone/Download https://gist.github.com/zsims/904dcb7d7fcc2543776740be65b171c0
 2. `yarn install` (or `npm install`)
 3. `yarn test` (or `npm test`)


## Relevant log files

```
$ yarn run test
yarn run v1.3.2
$ tsc
node_modules/@pact-foundation/pact-node/src/logger.ts(11,8): error TS2339: Property 'info' does not exist on type 'Logger'.
node_modules/@pact-foundation/pact-node/src/logger.ts(19,36): error TS2339: Property 'level' does not exist on type 'Logger'.
node_modules/@pact-foundation/pact-node/src/logger.ts(25,16): error TS2554: Expected 0 arguments, but got 1.
node_modules/@pact-foundation/pact-node/src/publisher.ts(10,3): error TS2322: Type '{}' is not assignable to type 'PublisherOptions'.
  Property 'pactFilesOrDirs' is missing in type '{}'.
node_modules/@pact-foundation/pact/dsl/verifier.d.ts(2,25): error TS2307: Cannot find module 'es6-promise'.
error Command failed with exit code 2.
```

--- a/package.json
+++ b/package.json
@@ -80,6 +80,7 @@
   "dependencies": {
     "@pact-foundation/pact-node": "^6.7.0",
     "cli-color": "^1.1.0",
+    "es6-promise": "^4.1.1",
     "lodash": "^4.17.4",
     "lodash.isfunction": "3.0.8",
     "lodash.isnil": "4.0.0",
@@ -112,7 +113,6 @@
     "chai-as-promised": "5.x",
     "coveralls": "^2.13.3",
     "enhanced-resolve": "^3.4.1",
-    "es6-promise": "^4.1.1",
     "imports-loader": "0.x",
     "istanbul": "0.4.x",
     "jasmine-core": "2.x",

